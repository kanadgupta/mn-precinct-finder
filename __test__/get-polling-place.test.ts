import { describe, expect, it } from 'vitest';

import precinctData from '../src/lib/data/mn-precincts.json' with { type: 'json' };
import pollingPlacesData from '../src/lib/data/mpls-2021-polling-places.json' with { type: 'json' };
import getPollingPlace from '../src/lib/get-polling-place.js';

describe('#getPollingPlace', () => {
  it('should retrieve polling place data for valid precinct name', () => {
    expect(getPollingPlace('Minneapolis W-1 P-3')).toMatchSnapshot();
  });

  it('should retrieve polling place data for valid precinct name with zeros', () => {
    expect(getPollingPlace('Minneapolis W-1 P-3')).toMatchSnapshot();
  });

  it('should return fallback for invalid precinct name', () => {
    expect(getPollingPlace('Minneapolis W-1000')).toBeNull();
  });
});

describe('polling place data', () => {
  const precinctKeys = Object.keys(pollingPlacesData);

  it.each(precinctKeys)('precinct %s exists in geojson data', key => {
    expect(
      precinctData.features.some(
        feature =>
          feature.properties.Precinct === key ||
          feature.properties.Precinct.replace(/W-0/g, 'W-').replace(/P-0/g, 'P-') === key,
      ),
    ).toBe(true);
  });

  it.todo('should check if all the Minneapolis precincts in the MN data exist in the polling place list');
});
