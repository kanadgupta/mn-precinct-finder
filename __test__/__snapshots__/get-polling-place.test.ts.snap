// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`#getPollingPlace > should retrieve polling place data for valid precinct name 1`] = `
{
  "address": "2955 Hayes St NE",
  "building": "Northeast Middle School",
  "directions": "enter via side/rear off 29th Ave NE",
  "gmapsUrl": "https://www.google.com/maps/search/?api=1&query=Northeast+Middle+School+2955+Hayes+St+NE&query_place_id=ChIJf6uXUNAts1IRx9gO5VPsm0M",
}
`;

exports[`#getPollingPlace > should retrieve polling place data for valid precinct name with zeros 1`] = `
{
  "address": "2955 Hayes St NE",
  "building": "Northeast Middle School",
  "directions": "enter via side/rear off 29th Ave NE",
  "gmapsUrl": "https://www.google.com/maps/search/?api=1&query=Northeast+Middle+School+2955+Hayes+St+NE&query_place_id=ChIJf6uXUNAts1IRx9gO5VPsm0M",
}
`;
